src/garage/experiment/experiment.py

```diff
@@ -223,64 +216,37 @@ def run_experiment(method_call=None,
-                   docker_image=None,
-                   aws_config=None,
                    env=None,
                    variant=None,
                    use_tf=False,
                    use_gpu=False,
-                   sync_s3_pkl=False,
-                   sync_s3_png=False,
-                   sync_s3_log=False,
-                   sync_log_on_termination=True,
-                   confirm_remote=True,
-                   terminate_machine=True,
-                   periodic_sync=True,
-                   periodic_sync_interval=15,
-                   sync_all_data_node_to_s3=True,
                    use_cloudpickle=None,
                    pre_commands=None,
-                   added_project_directories=[],
                    **kwargs):
-    """
-    Serialize the method call and run the experiment using the
+    """Serialize the method call and run the experiment using the
     specified mode.
-    :param method_call: A method call.
-    :param script: The name of the entrance point python script
-    :param mode: Where and how to run the experiment. Should be one of "local",
-     "local_docker", "ec2", or "lab_kube".
-    :param dry: Whether to do a dry-run, which only prints the commands without
-     executing them.
-    :param exp_prefix: Name prefix for the experiments
-    :param docker_image: name of the docker image. Ignored if using local mode.
-    :param aws_config: configuration for AWS. Only used under EC2 mode
-    :param env: extra environment variables
-    :param kwargs: All other parameters will be passed directly to the entrance
-     python script.
-    :param variant: If provided, should be a dictionary of parameters
-    :param use_tf: this flag is used along with the Theano and GPU
-     configuration when using TensorFlow
-    :param use_gpu: Whether the launched task is running on GPU. This triggers
-     a few configuration changes including
-    certain environment flags
-    :param sync_s3_pkl: Whether to sync pkl files during execution of the
-     experiment (they will always be synced at
-    the end of the experiment)
-    :param sync_s3_png: Whether to sync png files during execution of the
-     experiment (they will always be synced at
-    the end of the experiment)
-    :param sync_s3_log: Whether to sync log files during execution of the
-     experiment (they will always be synced at
-    the end of the experiment)
-    :param confirm_remote: Whether to confirm before launching experiments
-     remotely
-    :param terminate_machine: Whether to terminate machine after experiment
-     finishes. Only used when using mode="ec2". This is useful when one wants
-     to debug after an experiment finishes abnormally.
-    :param periodic_sync: Whether to synchronize certain experiment files
-     periodically during execution.
-    :param periodic_sync_interval: Time interval between each periodic sync,
-     in seconds.
+
+    Args:
+        method_call (object): A method call.
```

Comment:
are `method_call` and `batch_tasks` both callables?

Reply:
`batch_tasks` is a dict containing `method_call` and other info.
```python3
batch_tasks = [
            dict(
                kwargs,
                pre_commands=pre_commands,
                method_call=method_call,
                exp_name=exp_name,
                log_dir=log_dir,
                env=env,
                variant=variant,
                use_cloudpickle=use_cloudpickle)
        ]
```

Reply:
what is the type of method_call?

Reply:
Emmm it's a function. Shall I say `func` or `callable`?

Reply:
callable